### DISCUSSION TITLE
Get errors when trying to make on raspi

### DISCUSSION THREAD

**Discussion by volkerdeHaas** [wmbusmeters/wmbusmeters#1239] [Q&A]
I would like to use your genius software to collect values from 7 wMbus heat energy counters from Kampstrup (MultiCal 303). But when I cloned the project in a directory of my Raspi and tried to "make" I get this error (missing lib directory):

```
pi@DMXraspi01:~/GIT1/wmbusmeters $ make
New version number generates new build/version.h
Building 1.16.1-23-gefcc41d
g++ -Os -g -DFUZZING=false -fPIC -std=c++11 -Wall -Werror=format-security -Wno-unused-function -Ibuild src/address.cc -c -E > build/address.o.src
In file included from /usr/include/c++/10/string:40,
                 from src/util.h:23,
                 from src/address.h:21,
                 from src/address.cc:18:
/usr/include/c++/10/bits/char_traits.h:33:10: fatal error: fits/stl_alg/base.h: Datei oder Verzeichnis nicht gefunden
   33 | #include <fits/stl_alg/base.h>0 +/ s|t:copy, std::finl_n
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
I have udated and upgraded the system (bullseye) before trying to make. So Do I need to pre-install any other libs?

**Comment by avandorp** [comment]
I can't remember all the libraries I've needed, but libxml2-dev, pkg-config, librtlsdr-dev and libxslt1-dev were among the packages I've installed (possibly for other projects too).

**Reply by volkerdeHaas** [reply]
Thnx for your help. But I've already installed all of them. Seems to be a different lib. Maybe it's because I'm using raspian Bullseye and not Fedora?

**Reply by avandorp** [reply]
I'm on Debian stable, so I don't think so. EDIT: I've just looked at the test config (unfortunately running Ubuntu). It installs librtlsdr-dev libusb-dev libxml2-dev libxslt1-dev. You've run ./configure first?

**Reply by volkerdeHaas** [reply]
Yes I've done configure and that went okay. 
here is my pathway:
```
sudo apt-get update && sudo apt-get upgrade -y
sudo apt-get install libxml2-dev
sudo apt-get install pkg-config
sudo apt-get install librtlsdr-dev
sudo apt-get install libxslt1-dev
sudo apt-get install libusb-dev
sudo apt-get install mosquitto-clients 
sudo apt-get install git
git clone https://github.com/wmbusmeters/wmbusmeters.git
cd wmbusmeters
./configure
```
I get  this
`checking build system type... armv7l-unknown-linux-gnueabihf
checking host system type... armv7l-unknown-linux-gnueabihf
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking for rtlsdr_get_device_count in -lrtlsdr... yes
checking for xmlFreeDoc in -lxml2... yes
configure: creating /home/pi/GIT/wmbusmeters/build/armv7l-unknown-linux-gnueabihf/config.status
config.status: creating /home/pi/GIT/wmbusmeters/build/armv7l-unknown-linux-gnueabihf/spec.gmk
config.status: creating /home/pi/GIT/wmbusmeters/build/armv7l-unknown-linux-gnueabihf/Makefile
`
Then I do
make
getting
`Building 1.16.1-23-gefcc41d
g++ -Os -g -DFUZZING=false -fPIC -std=c++11 -Wall -Werror=format-security -Wno-unused-function -Ibuild src/address.cc -c -E > build/address.o.src
In file included from /usr/include/c++/10/string:40,
                 from src/util.h:23,
                 from src/address.h:21,
                 from src/address.cc:18:
/usr/include/c++/10/bits/char_traits.h:33:10: fatal error: fits/stl_alg/base.h: Datei oder Verzeichnis nicht gefunden
   33 | #include <fits/stl_alg/base.h>0 +/ s|t:copy, std::finl_n
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:147: build/address.o] Fehler 1
`

**Reply by avandorp** [reply]
I haven't compiled on arm for some time. I do use Bookworm and not old-stable though.
Weirdly enough my char_traits.h includes a <bits/stl_algobase.h> and not <fits/stl_alg/base.h>.

**Comment by volkerdeHaas** [comment]
I flashed a complete new sd card with buster image. This solved the problem. May be my bullseye was corrupt or it only works with older versions.

**Reply by avandorp** [reply]
Both b to f and o to / are one-bit-flips. I'd discard that old flash card.

**Comment by volkerdeHaas** [comment]
possibly defect or corrupted operating system
